ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java

```diff
@@ -114,6 +120,14 @@ protected void setAccumulator(final BonsaiWorldStateUpdateAccumulator accumulato
     this.accumulator = accumulator;
   }
 
+  @Override
+  public Optional<WorldStateProof> getAccountProof(
+      final Hash worldStateRoot,
+      final Address accountAddress,
+      final List<UInt256> accountStorageKeys) {
+    return worldStateProof.getAccountProof(worldStateRoot, accountAddress, accountStorageKeys);
```

Comment:
should we use this.worldStateRootHash instead?  passing the state root makes sense for forest, but not really for bonsai.

Reply:
you right I modified the code, I moved the getProof to archive again with some modification. it will normally makes sense now . if you can review again